helper/fileperms/fileperms.go

```diff
@@ -0,0 +1,71 @@
+package fileperms
+
+import (
+	"fmt"
+	"os"
+)
+
+// mode represents a simplified octal representation of a Unix file permissions
+// mode interface provided by os.FileMode.
+//
+// This package is intended to prevent accidental incorrect file permissions caused
+// by erroneous octal representations of permissions bits in Go's decimal base. A
+// linter can be configured to assert calls to functions such as os.OpenFile,
+// os.Chmod, os.File.Chmod, etc. are passing a parameter of type fileperms.mode.
+// Anything else will be flagged and must be changed. By not exporting the mode
+// type, getting past the linter means funneling through one of the exported const
+// values in this package.
+//
+// For example, the code `os.Chmod(755)` produces file permissions "-wxrw--wt",
+// whereas the intended octal form `os.Chmod(0755)` produces "rwxr-xr-x".
+type mode = os.FileMode
+
+// 0) no permission
+// 1) execute
+// 2) write
+// 3) execute + write
+// 4) read
+// 5) read + execute
+// 6) read + write
+// 7) read + write + execute
+const (
+	Oct555 mode = 0555
```

Comment:
I like the idea of moving these to consts, but rather than naming them after the octal values, could we take advantage of the wrapper function to give us a more literate API? (i.e. something more like `os.MkdirAll(path, "rwxrwxrwx")`

Reply:
Yeah I think that's a good idea - but probably worth bikeshedding more in the other ticket. I'm gonna merge this just to get these bugs fixed in time, but we can definitely tweak afterwords.

Reply:
I'd settle for meaningful const names like:
WorldR = 555
OwnerRW = 600
(610 doesn't make any sense AFAICT and the test should be changed to something sensible)
OwnerRWXWorldRX = 655
OwnerGroupRW = 660
WorldRW = 666
OwnerRWXWorldRX = 755
WorldRWX = 777

But no biggie. Your approach solves the immediate problem.